src/core/thermostat.hpp

```diff
@@ -94,8 +91,13 @@ extern bool thermo_virtual;
  * parameter structs
  ************************************************/
 
+struct BaseThermostat {
```

Comment:
Can you please explain why the base class is needed? As far as I can tell the inheritance is not used, and in general (mutable) data members in base classes are a bad idea (due to the non-locality of mutation).

Reply:
I originally thought about moving most of the thermostat RNG MPI callbacks into the base struct as getters and setters, and only have a few MPI callbacks (not macro-generated) taking as argument the thermostat switch to select from which global object to call the setter/getter, but these plans didn't materialize. I wouldn't mind reverting that inheritance; it's probably too early to look into that yet.

If by non-locality of mutation you mean the ability of Brownian to change the value of the Langevin counter, I'm not sure how we could run into that situation, as having a `BaseThermostat` method that takes a ref to another `BaseThermostat` object is not useful.